torch/distributed/rpc/api.py

```diff
@@ -259,35 +264,57 @@ def remote(to, func, args=None, kwargs=None):
     kwargs = kwargs if kwargs else {}
 
     info = _to_worker_info(to)
+    current_worker = get_worker_info()
+    # If profiling is enabled, kick off the timer and retrieve back a
+    # RecordFunction instance.
+    rf = None
+    if torch.autograd._profiler_enabled():
+        rf = _start_record_function(
+            RPCExecMode.REMOTE,
+            str(qualified_name if qualified_name is not None else func.__name__),
+            current_worker.name,
+            info.name,
+        )
     if qualified_name is not None:
         return _invoke_remote_builtin(
-            _agent, info, qualified_name, *args, **kwargs)
+            _agent, info, qualified_name, rf, *args, **kwargs)
     else:
         (pickled_python_udf, tensors) = _default_pickler.serialize(
             PythonUDF(func, args, kwargs))
         return _invoke_remote_python_udf(
-            _agent, info, pickled_python_udf, tensors)
+            _agent, info, pickled_python_udf, tensors, rf)
 
 
-def _invoke_rpc(to, func, args=None, kwargs=None):
+def _invoke_rpc(to, func, rpc_type, args=None, kwargs=None):
     if not callable(func):
         raise TypeError("function should be callable.")
 
     qualified_name = torch.jit._find_builtin(func)
+    info = _to_worker_info(to)
+    current_worker = get_worker_info()
+    # If profiling is enabled, kick off the timer and retrieve back a
+    # RecordFunction instance.
+    rf = None
+    if torch.autograd._profiler_enabled():
+        rf = _start_record_function(
```

Comment:
Should be the first line in this function.

Reply:
Same issue, we need to pass the `qualified_name` into the `_start_record_function` call, so that we can call `RecordFunction::before` with the right key.